src/app/people/daos/peopleDAO.py

```diff
@@ -136,6 +136,37 @@ def find_people_with_birthday_before_given_date(self, date: datetime.date) -> Li
 
         return query.all()
 
+    def find_people_with_anniversary_before_given_date(self, given_date: datetime.date) -> list[models.Person]:
```

Comment:
@ItuC300 i added a lot of changes to the birthday method after the one below. did you take a look at those changes?

Reply:
We should work on the unit tests first in this case, since it should be a simple copy paste from the birthday ones, and then just using the anniversary date instead of the birthdate. Then you can work on improving the method until all the tests pass. basically TDD :).